python/JSBSim.py

```diff
@@ -108,6 +120,26 @@ def CheckXMLFile(f):
 
 fdm.run_ic()
 fdm.print_simulation_configuration()
-
-while fdm.run() and fdm.get_sim_time() <= args.end:
-    pass
+frame_duration = fdm.get_delta_t()
+sleep_nseconds = (frame_duration if args.realtime else sleep_period) * 1E9
+current_seconds = initial_seconds = time.time()
+
+while fdm.get_sim_time() <= args.end:
+    if args.realtime:
+        current_seconds = time.time()
+        actual_elapsed_time = current_seconds - initial_seconds
+        sim_lag_time = actual_elapsed_time - fdm.get_sim_time()
+
+        for i in range(0, int(sim_lag_time / frame_duration)):
+            fdm.run()
+            cycle_duration = time.time() - current_seconds
+            fdm.set_property_value("simulation/cycle_duration", cycle_duration)
+
+            current_seconds = time.time()
+            if fdm.holding():
+                break
+    else:
+        fdm.run()
```

Comment:
Same as above, the result should be stored in a variable `result` that must be checked by the `while` loop
```suggestion
        result = fdm.run()
```